src/java/src/main/java/com/dnanexus/DXSearch.java

```diff
@@ -900,11 +909,11 @@ public int size() {
         /**
          * Iterator implementation for findDataObjects results.
          */
-        private class ResultIterator
+        public class ResultIterator
```

Comment:
I tried to make **ResultIterator** private and labeled as *@VisibleForTesting*, but compiler claimed about that:

```
        @VisibleForTesting
        private class ResultIterator
...
[INFO] Compilation failure
/vagrant/src/java/src/test/java/com/dnanexus/DXSearchTest.java:[596,74] error: FindDataObjectsResult.ResultIterator has private access in FindDataObjectsResult
```

If to leave **ResultIterator** public and to make **int pageNo()** method private compiler could not find a respective symbol:

```
        @VisibleForTesting
        private int pageNo() {
            return this.currentPageNo;
        }
...
[INFO] Compilation failure
/vagrant/src/java/src/test/java/com/dnanexus/DXSearchTest.java:[596,95] error: cannot find symbol
```
I tried to label a **PaginatingFindResultIterator** as *@VisibleForTesting*, but it did not help.

Will look to that tomorrow.